### DISCUSSION TITLE
Httpx returns short result in python

### DISCUSSION THREAD

**Discussion by HackerEunji** [projectdiscovery/httpx#1684] [General]
Hello everyone. Now I'm using httpx in python but t he commands I used are as follows.

```
import os

os.system("httpx -l all_contents.txt -o result.txt") 
```
I want to collect possible domains into a txt file and make sure it works properly on httpx. Other than that command, I'm running several commands together through os.system, but httpx is the only problem. Running a Python file returns ridiculously short results compared to running httpx commands in the terminal window. The all_contents.txt file contains multiple domains.

**In the python code**
```
                                            httpx Ascii logo here 
                                                                       projectdiscovery.io 
                         
                                    [INF] Current httpx version v1.6.0 (latest) 
                                           https://107one.000.com 
                                           https://id.america.000.com 
```
only 2 result returns.

But,
**In the terminal**
```
                                            httpx Ascii logo here 
                                                                       projectdiscovery.io 
                         
                                    [INF] Current httpx version v1.6.0 (latest) 
                                           https://107one.000.com 
                                           https://id.america.000.com 
                                           https://ab.fff.000.com
                                           https://df.www.000.com
                                                        .
                                                        .
                                                        .
```

The results are much longer. Over 30 domains emerge.
If httpx doesn't work at all, I definitely think I misused the code. But in this case, the problem is that httpx works well, but the execution results don't work for long.My guess is, "There is a difference in the output of the results between the terminal and the execution of the Python code." When I run the Python code, it seems that something interferes with the output, or I don't get all the results executed in the window. But I don't know the exact reason.

What I tried was:

**1) Change os.system to subprocess Popen command**

```
import subprocess 

command = "httpx -l all_contents.txt -o result.txt" 
process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE) 
stdout, stderr = process.communicate() 
print("STDOUT:", stdout.decode()) 
print("STDERR:", stderr.decode()) 
```

This returns short results, as is the case with os.system, and when I output an error, the command returns 0 that means operating normally.

**2) Change os.system to subprocess check_output command**
```
import os 

command = "httpx -l all_contents.txt -o result.txt" 
result = subprocess.check_output(command, shell=True, text=True) 
print(result) 
```
It runs httpx, but it doesn't even return any results.

**3) Adjusting options for httpx**
```
import os 

os.system("httpx -l all_contents.txt -o result.txt -t 100 -timeout 5") 
```
I thought maybe concurrent requests were a problem, so I adjusted the thread as well. This is also useless.

**4) Adding Environmental Variables**
```
import os 

os.environ['something'] = 'example_value'
```
In this case, I've tried to go this far, but I haven't found the correct command yet. 

**5) Execute the command in new terminal**
```
import os 

command = f'xterm -e "httpx -l all_contents.txt -o result.txt ; exit"'
subprocess.run(command, shell=True)
```
I even thought it was a Python problem, so I even opened a new external terminal called xterm and ran the command. But the same problem happens here! Because of this, I definitely think it's a problem with httpx. Reinstall httpx didn't also help.

**Additional Error Occured ########################################**
```
httpx -l all_contents.txt -o result.txt -timeout 2000 -follow-redirects -v
```

```
httpx version check failed: [updater:RUNTIME] http Get https://api.pdtm.sh/api/v1/tools/httpx?arch=amd64&go_version=go1.22.2&machine_id=45c2bde64e9244c1faf1e2f0f852ad6825e94779171b9ee8a41c05816da610fa&os=linux&utm_source=unknown&v=v1.6.0 failed <- Get "https://api.pdtm.sh/api/v1/tools/httpx?arch=amd64&go_version=go1.22.2&machine_id=45c2bde64e9244c1faf1e2f0f852ad6825e94779171b9ee8a41c05816da610fa&os=linux&utm_source=unknown&v=v1.6.0": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```
While receiving short results as described above, the above error occurred anew. Is this a problem with the network?

I've tried numerous ways, but I haven't solved them. I need your help

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
